s3filepath/s3filepath.go

```diff
@@ -68,94 +34,47 @@ type S3File struct {
 	ConfFile  string
 }
 
+// PathChecker is the interface for determining if a path in S3 exists, which allows
+// DI for testing.
+type PathChecker interface {
+	FileExists(path string) bool
+}
+
+// S3PathChecker will use pathio to determine if the path actually exists in S3, and
+// will be used in prod.
+type S3PathChecker struct{}
+
+// FileExists looks up if the file exists in S3 using the pathio.Reader method.
+func (S3PathChecker) FileExists(path string) bool {
+	_, err := pathio.Reader(path)
```

Comment:
hmm - I wonder if this would end up leaking file handles, I don't know enough about Pathio though. At least in the way we intend this worker to be run, it shouldn't be a big deal, but something to consider...